src/librustc_mir/transform/const_prop.rs

```diff
@@ -852,20 +856,34 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
                     if let Some(()) = self.const_prop(rval, place_layout, source_info, place) {
                         if can_const_prop == ConstPropMode::FullConstProp
                             || can_const_prop == ConstPropMode::OnlyPropagateInto
+                            || can_const_prop == ConstPropMode::OnlyInsideOwnBlock
```

Comment:
This is fine but we could also just say `can_const_prop != ConstPropMode::NoPropagation`

Reply:
Indeed!

Reply:
(I changed it to that and now the intent is clearer <3)